mito-ai/mito_ai/file_uploads/handlers.py

```diff
@@ -6,7 +6,40 @@
 import tornado
 from typing import Dict, Any
 from jupyter_server.base.handlers import APIHandler
-from mito_ai.utils.telemetry_utils import log_file_upload_attempt, log_file_upload_failure
+from mito_ai.utils.telemetry_utils import (
+    log_file_upload_attempt,
+    log_file_upload_failure,
+)
+
+MAX_IMAGE_SIZE_MB = 3
+
+
+def _is_image_file(filename: str) -> bool:
+    image_extensions = {
+        ".jpg",
+        ".jpeg",
+        ".png",
+        ".gif",
+        ".bmp",
+        ".tiff",
+        ".tif",
+        ".webp",
+        ".svg",
+    }
+    file_extension = os.path.splitext(filename)[1].lower()
+    return file_extension in image_extensions
+
+
+def _check_image_size_limit(file_data: bytes, filename: str) -> None:
+    if _is_image_file(filename):
```

Comment:
Most minimal thing ever, but btw I think writing checks like this is a bit nicer because it removes a layer of nesting. No need to change now, but just a little pointer if you want it for the future. 

```python
if not _is_image_file(filename):
     return 

file_size_mb = ...
```